sdk/lib/opentelemetry/sdk/configurator.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    # The configurator provides defaults and facilitates configuring the
+    # SDK for use.
+    class Configurator
+      USE_MODE_UNSPECIFIED = 0
+      USE_MODE_ONE = 1
+      USE_MODE_ALL = 2
+
+      private_constant :USE_MODE_UNSPECIFIED, :USE_MODE_ONE, :USE_MODE_ALL
+
+      attr_writer :tracer_factory, :logger
+
+      def initialize
+        @adapter_names = []
+        @adapter_config_map = {}
+        @span_processors = []
+        @use_mode = USE_MODE_UNSPECIFIED
+      end
+
+      def tracer_factory
+        @tracer_factory ||= Trace::TracerFactory.new
```

Comment:
Is it necessary to support replacing the tracer factory? I can see you'd want to do that in order to configure the tracer factory, but the documentation earlier suggests replacing the implementation.

Reply:
I suppose this doesn't make a whole lot of sense. It seems like a user would be using a different SDK, or a hybrid SDK of sorts. In either case, we don't have to support that for the default SDK, and I don't think we should, unless a use case becomes apparent.